mmdet/core/evaluation/eval_hooks.py

```diff
@@ -37,9 +63,45 @@ def __init__(self, dataloader, start=None, interval=1, **eval_kwargs):
         self.dataloader = dataloader
         self.interval = interval
         self.start = start
+        self.save_best = save_best
         self.eval_kwargs = eval_kwargs
         self.initial_epoch_flag = True
 
+        self.logger = get_root_logger()
+
+        if self.save_best is not None:
+            self._init_rule(rule, self.save_best)
+            self.best_json = dict()
+
+    def _init_rule(self, rule, key_indicator):
+        """Initialize rule, key_indicator, comparison_func, and best score.
+
+        Args:
+            rule (str | None): Comparison rule for best score.
+            key_indicator (str | None): Key indicator to determine the
+                comparison rule.
+        """
+        if rule not in self.rule_map and rule is not None:
+            raise KeyError(f'rule must be greater, less or None, '
+                           f'but got {rule}.')
+
+        if rule is None:
+            if key_indicator != 'auto':
+                if any(key in key_indicator for key in self.greater_keys):
+                    rule = 'greater'
+                elif any(key in key_indicator for key in self.less_keys):
+                    rule = 'less'
+                else:
+                    raise ValueError(
+                        f'key_indicator must be in {self.greater_keys} '
+                        f'or in {self.less_keys} when rule is None, '
+                        f'but got {key_indicator}')
```

Comment:
Make the message clearer:
Cannot infer the rule for key {key_indicator}, thus a specific rule must be specified.